src/Sarif.Viewer.VisualStudio/Models/FileDetailsModel.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Text;
+
+namespace Microsoft.Sarif.Viewer.Models
+{
+    public class FileDetailsModel
+    {
+        /// <summary>
+        /// Contents of file. May or may not be Base64 encoded.
+        /// </summary>
+        private string _contents;
+
+        /// <summary>
+        /// Tells whether or not the _contents are Base64 encoded.
+        /// </summary>
+        private bool _encoded;
+
+        /// <summary>
+        /// Creates a new object of the <see cref="FileDetailsModel" /> class.
+        /// </summary>
+        /// <param name="hash">SHA256 hash for file.</param>
+        /// <param name="contents">Base64 encoded contents of the file.</param>
+        public FileDetailsModel(string hash, string contents)
+        {
+            Hash = hash;
+            _contents = contents;
+            _encoded = true;
+        }
+
+        /// <summary>
+        /// SHA256 hash for file.
+        /// </summary>
+        public string Hash { get; }
+
+        /// <summary>
+        /// Returns the decoded file contents.
+        /// </summary>
+        public string Contents
+        {
+            get
+            {
+                // If the contents are encoded, decode them.
+                if (_encoded)
+                {
+                    DecodeContents();
```

Comment:
>DecodeContents [](start = 20, length = 14)

Decoding is a non-trivial operation and so typically wouldn't be a good candidate for a property accessor. If you accessed this member in a debug watch window, for example, you will change the state of the object. Ouch.

A GetContents method would therefore be preferable.

Now, you could also consider decoding on instantiating this object, and save a lot of bother. If that has adverse performance impact, however, not a good idea (but if in practical use we can't defer decoding costs, we should consider it).

If my suggestion to prefer a GetContents method causes issues in WPF binding, that's a good reason to ignore my comment, so feel free to push back.